src/components/surveys/WordCloudChart.vue

```diff
@@ -0,0 +1,717 @@
+<template>
+  <div class="word-cloud-container">
+    <div v-if="words.length > 0" ref="cloudContainer" class="word-cloud">
+      <div
+        v-for="(word, index) in words"
+        :key="word.text"
+        class="word"
+        :style="{
+          fontSize: `${getWordSize(word.count)}px`,
+          color: getWordColor(word.count),
+          opacity: wordPositions[index]?.x === -9999 ? 0 : getWordOpacity(word.count),
+          position: 'absolute',
+          left: `${wordPositions[index]?.x || 0}px`,
+          top: `${wordPositions[index]?.y || 0}px`,
+          transform: 'translate(-50%, -50%)',
+          padding: '4px',
+          cursor: 'pointer',
+          zIndex: word.count === selectedWord?.count ? 10 : 1,
+          display: wordPositions[index]?.x === -9999 ? 'none' : 'block',
+        }"
+        @click="toggleWordSelection(word)"
+      >
+        {{ word.text }}
+      </div>
+    </div>
+    <div v-else class="no-data">
+      <p>No text data available</p>
+    </div>
+
+    <div v-if="selectedWord" class="word-tooltip">
+      <div class="tooltip-header">
+        <strong>{{ selectedWord.text }}</strong>
+        <span class="count-badge">{{ selectedWord.count }}</span>
+      </div>
+      <div v-if="selectedWord.fullResponses && selectedWord.fullResponses.length > 0" class="tooltip-responses">
+        <div v-for="(response, index) in selectedWord.fullResponses.slice(0, 3)" :key="index" class="response-item">
+          <div class="response-text">{{ response }}</div>
+          <div v-if="index < Math.min(selectedWord.fullResponses.length - 1, 2)" class="response-divider"></div>
+        </div>
+        <div v-if="selectedWord.fullResponses.length > 3" class="more-responses">
+          And {{ selectedWord.fullResponses.length - 3 }} more responses...
+        </div>
+      </div>
+      <button class="close-button" @click="selectedWord = null">×</button>
+    </div>
+  </div>
+</template>
+
+<script lang="ts">
+import { defineComponent, ref, computed, PropType, onMounted, reactive, watch } from "vue";
+import cloud from "d3-cloud";
```

Comment:
Use charjs and not d3-cloud https://www.npmjs.com/package/chartjs-chart-wordcloud